app/k9mail/src/test/java/com/fsck/k9/account/FakeAccountManager.kt

```diff
@@ -7,7 +7,8 @@ import com.fsck.k9.preferences.AccountManager
 import kotlinx.coroutines.flow.Flow
 
 class FakeAccountManager(
-    private val accounts: Map<String, Account> = emptyMap(),
+    private val accounts: MutableMap<String, Account> = mutableMapOf(),
```

Comment:
[Optional] Using a `Map` for the "initial accounts" parameter and a `MutableMap` as a hidden implementation detail would lead to better encapsulation. And existing call sites wouldn't have to be changed.

Reply:
This is a test fake and I'm not sure if it needs more encapsulation. Also why is changing call sites an issue for you?

Reply:
Test code is code. It still needs to be maintained. That's generally easier if implementation details are not leaked.

It's not so much about changing call sites, it's about the time it requires to change call sites. If there's a benefit to doing so, sure go for it. If there's an easy way to avoid it, why not use it? It usually also means less changed lines to review. In this case it's not really an issue. But in cases where there are 20+ call sites it probably does.

Looking at this again, I'd like to change my (still optional) suggestion: Callers shouldn't have to care that `FakeAccountManager` is using a `(Mutable)Map` internally. If a caller needs to provide initial accounts, a list would do:

```kotlin
val accountManager = FakeAccountManager(accounts = mutableMapOf(ACCOUNT_UUID to createAccount(ACCOUNT_UUID)))

// vs.

val accountManager = FakeAccountManager(initialAccounts = listOf(createAccount(ACCOUNT_UUID)))
```

Reply:
There is no doubt that test code needs the same level of detail as non test code. But I doubt that this level of detail is benefitial taking into account the projects goals and existing quality.